source/PlayerInfo.cpp

```diff
@@ -2354,7 +2363,37 @@ void PlayerInfo::StepMissions(UI *ui)
 		else if(mission.CanComplete(*this))
 			RemoveMission(Mission::COMPLETE, mission, ui);
 		else if(mission.Destination() == GetPlanet() && !freshlyLoaded)
+		{
 			mission.Do(Mission::VISIT, *this, ui);
+			if(mission.IsUnique())
+				continue;
+			
+			// On visit dialogs are handled separately as to avoid a player
+			// getting spammed by on visit dialogs if they are stacking jobs
+			// from the same destination.
+			auto ait = mission.GetAction(Mission::VISIT);
+			if(!ait.DialogText().empty())
+			{
+				if(visitText.empty())
+					visitText = Format::Replace(ait.DialogText(), subs);
+				else
+					++extraVisitDialogs;
+				if(mission.Deadline())
+					++deadlineMissions;
+			}
+		}
+	}
+	if(!visitText.empty())
+	{
+		if(extraVisitDialogs)
+		{
+			visitText += "\n\tYou have " + Format::Number(extraVisitDialogs) + " other unfinished " + ((extraVisitDialogs > 1) ? "missions" : "mission") + " at this location.";
+			if(deadlineMissions)
+			{
+				visitText += "\n\t" + Format::Number(deadlineMissions) + " of your unfinished missions here " + ((deadlineMissions > 1) ? "have approaching deadlines." : "has an approaching deadline.");
```

Comment:
Is there a better construct that doesn't start a sentence with a digit?

Reply:
I would like one.
"Of your unfinished missions here, X have approaching deadlines."?

Reply:
I'd be open to having the deadline missions be just an parenthesized addendum, e.g.

"You have 8 other unfinished missions here (4 of which have deadlines)."